backend/RestAPI/Controllers/VideosController.cs

```diff
@@ -158,10 +159,55 @@ public async Task<ActionResult<FileStreamResult>> Stream(Guid videoId, Guid user
                 return Unauthorized();
             }
 
-            var response = File(System.IO.File.OpenRead(video.Path), "video/mp4", enableRangeProcessing: true);
+            var response = File(System.IO.File.OpenRead(video.Path), "video/mp4",video.Title + Path.GetExtension(video.Path),enableRangeProcessing: true);
```

Comment:
I dont think you need to explicitly specify enableRangeProcessing here

Reply:
oh yeah, right, skipped it after cleaning up